test/e2e/framework/expect.go

```diff
@@ -90,3 +90,23 @@ func ExpectHaveKey(actual interface{}, key interface{}, explain ...interface{})
 func ExpectEmpty(actual interface{}, explain ...interface{}) {
 	gomega.ExpectWithOffset(1, actual).To(gomega.BeEmpty(), explain...)
 }
+
```

Comment:
Why do we need to add the following functions?
Those functions are not used in this pull request.

Reply:
In my initial attempt to resolve this flake, I was going to made amendments to the device plugin e2e test by checking that the number of pods are greater than 2 instead of strict equality as in some cases there was a log rotation pod unexpectedly still running on the node. I later pivoted to an approach of making sure that the log rotation pods are cleaned up properly. 

In spite of the change of approach, I thought it would be useful to leave the functions that I added as `expect.go` is a wrapper around gomega matcher and we could benefit from some additional functions that provide us the ability to perform more granular comparisons in the future. 

I deliberately kept it as a separate commit in case it is considered unnecessary and needs to be removed.

Reply:
This looks like a nice and sensible cleanup step for the device plugin test.  I see that one of the new functions is used in this test: `ExpectEqual`.  It does seem a bit out-of-place to put the other new functions in this PR, but I do like them.  I am in favor of keeping them and getting this PR going to reduce the flake count.